### ISSUE TITLE
feature or documentation request: way to redirect in webadmin handler

### ISSUE THREAD

**Issue by curioussavage** [cuberite/cuberite#3263]
So a common thing for a web server to do is to redirect the user after a form submit to a new path on the site. Is this possible in cuberite? If not, would this be hard to enable?

**Comment by NiLSPACE** [comment]
This should go into the Core instead of Cuberite.

Also, yes, it's possible, and shouldn't be too hard to do.

**Comment by madmaxoft** [comment]
@NiLSPACE I don't think so. This request is so that webadmin handlers may return the HTTP status 302 with a new Location, which is currently impossible.

**Comment by NiLSPACE** [comment]
I guess that is true, but the same can be done by putting in a meta tag that redirects or a small JS script.

**Comment by madmaxoft** [comment]
Not really, because anything like that registers in the browser's history, so it's impossible to use the Back button properly afterwards.

**Comment by curioussavage** [comment]
Should I still move the issue to core? I put it here because it seemed more than trivial as @madmaxoft pointed out.

As far as api, if this were implemented it would be nice to have a cPluginResponse class or whatever you want to call it, that you can set the body on and that would have a redirect method on it. 

```lua
-- either set body and return
Response.body = templateString
return Response
-- or return and call redirect
return Response.redirect(302, pathToDesiredPage)
```

**Comment by madmaxoft** [comment]
No, don't move it, it has nothing to do with the Core plugin.

As for the API, we need to stay compatible with existing plugins, there's already too many of them using the API, we can't afford to break it.

I've had plans for a webadmin overhaul, which included the possibility to send completely custom HTTP responses (including the HTTP status code and headers), as well as continuation ("here's a chunk of the page, I'll give you the rest later"), but those have been moved to the background by more pressing matters.

**Comment by curioussavage** [comment]
ah! your right. I wasn't even thinking about backwards compatibility
